go/database/mpt/rlp/rlp.go

```diff
@@ -253,3 +364,45 @@ func (i BigInt) getEncodedLength() int {
 	length := ((bitlen + 7) & -8) >> 3
 	return getEncodedLengthLength(length) + length
 }
+
+// readNumber interprets the first 'slen' bytes of 'b' as a big-endian integer.
+// It returns the resulting integer and any error encountered.
+//
+// The function expects 'slen' to be the size of the integer to be read from 'b'.
+// If 'slen' is greater than the length of 'b', an error is returned.
+//
+// The function supports reading integers of up to 8 bytes (64 bits).
+// The integer is constructed by shifting and OR-ing the bytes from 'b'.
```

Comment:
nit: this line is an implementation detail

Reply:
right, removed